src/main/java/io/stargate/sgv2/jsonapi/service/operation/MetadataAttempt.java

```diff
@@ -0,0 +1,275 @@
+
+    } else if (columnMetadata.getType()
+        instanceof com.datastax.oss.driver.api.core.type.SetType st) {
+      if (!st.isFrozen()) {
+        final Optional<ApiDataTypeDef> apiDataTypeDef = ApiDataTypeDefs.from(st.getElementType());
+        if (apiDataTypeDef.isPresent()) {
+          return new ComplexTypes.SetType(
+              PrimitiveTypes.fromString(apiDataTypeDef.get().getApiType().getApiName()));
+        }
+      }
+      // return unsupported format
+      return new ComplexTypes.UnsupportedType(st.asCql(true, false));
+    } else {
+      final Optional<ApiDataTypeDef> apiDataTypeDef =
+          ApiDataTypeDefs.from(columnMetadata.getType());
+      if (apiDataTypeDef.isPresent())
+        return PrimitiveTypes.fromString(apiDataTypeDef.get().getApiType().getApiName());
+      else {
+        // Need to return unsupported type
+        return new ComplexTypes.UnsupportedType(columnMetadata.getType().asCql(true, false));
+      }
+    }
+  }
+
+  /**
+   * Object used to build the response for listTables command
+   *
+   * @param name
+   * @param definition
+   */
+  @JsonPropertyOrder({"name", "definition"})
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  public record TableResponse(String name, TableDefinition definition) {
+
+    @JsonPropertyOrder({"columns", "primaryKey"})
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    record TableDefinition(Map<String, ColumnType> columns, PrimaryKey primaryKey) {}
+  }
+
+  protected List<TableSchemaObject> getTables() {
+    return keyspaceMetadata
+        .get()
+        // get all tables
+        .getTables()
+        .values()
+        .stream()
+        // filter for valid collections
+        .filter(TABLE_MATCHER.negate())
+        // map to name
+        .map(table -> TableSchemaObject.from(table, objectMapper))
+        // get as list
+        .toList();
+  }
+
+  /**
+   * EmptyAsyncResultSet implementation to be used only for metadata attempt where no cql query is
+   * run.
+   */
+  private static class EmptyAsyncResultSet implements AsyncResultSet {
+    @NonNull
+    @Override
+    public ColumnDefinitions getColumnDefinitions() {
+      return EmptyColumnDefinitions.INSTANCE;
+    }
+
+    @NonNull
+    @Override
+    public ExecutionInfo getExecutionInfo() {
+      return null;
+    }
+
+    @NonNull
+    @Override
+    public Iterable<Row> currentPage() {
+      return Collections.emptyList();
+    }
+
+    @Override
+    public int remaining() {
+      return 0;
+    }
+
+    @Override
+    public boolean hasMorePages() {
+      return false;
+    }
+
+    @NonNull
+    @Override
+    public CompletionStage<AsyncResultSet> fetchNextPage() throws IllegalStateException {
+      throw new IllegalStateException(
+          "No next page. Use #hasMorePages before calling this method to avoid this error.");
+    }
+
+    @Override
+    public boolean wasApplied() {
+      return true;
+    }
+  }
+
+  public static class NoRetryPolicy extends RetryPolicy {
```

Comment:
This is the same as OperationAttempt.RetryPolicy.NO_RETRY but changes the default from 1 ms to 10 ms , is there a reason not to use the existing ? 

also, if all it needs is to change the retry time create an instance of the OperationAttempt.RetryPolicy